homeassistant/components/automation/__init__.py

```diff
@@ -467,7 +473,8 @@ async def async_disable(self):
             self._async_detach_triggers()
             self._async_detach_triggers = None
 
-        await self.action_script.async_stop()
+        if stop_actions:
+            await self.action_script.async_stop()
```

Comment:
Should this happen before or after we update the state? I would say we can first write ha state, because the triggers are no longer attached. User can still see something is active by looking at `runs` attribute?

Reply:
This is cancelling the runs, which affects the state (i.e., the `current` attribute which comes from `self.action_script.runs` in `state_attributes()`.) So I would say it needs to happen before updating the state, like it is now.